Makefile

```diff
@@ -7,13 +7,19 @@
 # ----------------------------------------------------------------------------
 
 ifeq ($(WITH_COVERAGE), TRUE)
-	TEST_COMMAND = python setup.py test -a "--cov=skbio"
+	TEST_COMMAND = coverage run --rcfile ../.coveragerc -m skbio.test; coverage report --rcfile ../.coveragerc
 else
-	TEST_COMMAND = python setup.py test
+	TEST_COMMAND = python -m skbio.test
 endif
 
+# cd into a directory that is different from scikit-bio root directory to
+# simulate a user's install and testing of scikit-bio. Running from the root
+# directory will find the `skbio` subpackage (not necessarily the installed
+# one!) because cwd is considered in Python's search path. It is important to
+# simulate a user's install/test process this way to find package data that did
+# not install correctly (for example).
 test:
-	$(TEST_COMMAND)
+	cd ci && $(TEST_COMMAND); cd -
```

Comment:
I don't think this is necessary, since the `--pyargs` flag in pytest handles the python search path for us (we use this in QIIME 2, too). Testing locally without this appeared to work as expected.

Reply:
Oddly enough when I tested locally things worked fine (as you mention), but when the build was executed in Travis this didn't work. I kept seeing this exception:

```python
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/env_name/lib/python3.6/runpy.py", line 183, in _run_module_as_main
    mod_name, mod_spec, code = _get_module_details(mod_name, _Error)
  File "/home/travis/miniconda/envs/env_name/lib/python3.6/runpy.py", line 109, in _get_module_details
    __import__(pkg_name)
  File "/home/travis/build/biocore/scikit-bio/skbio/__init__.py", line 12, in <module>
    import skbio.io  # noqa
  File "/home/travis/build/biocore/scikit-bio/skbio/io/__init__.py", line 242, in <module>
    import_module('skbio.io.format.clustal')
  File "/home/travis/miniconda/envs/env_name/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/home/travis/build/biocore/scikit-bio/skbio/io/format/clustal.py", line 148, in <module>
    from skbio.alignment import TabularMSA
  File "/home/travis/build/biocore/scikit-bio/skbio/alignment/__init__.py", line 203, in <module>
    from ._tabular_msa import TabularMSA
  File "/home/travis/build/biocore/scikit-bio/skbio/alignment/_tabular_msa.py", line 17, in <module>
    from skbio.metadata._mixin import MetadataMixin, PositionalMetadataMixin
  File "/home/travis/build/biocore/scikit-bio/skbio/metadata/__init__.py", line 27, in <module>
    from ._interval import Interval, IntervalMetadata
  File "/home/travis/build/biocore/scikit-bio/skbio/metadata/_interval.py", line 13, in <module>
    from ._intersection import IntervalTree
ModuleNotFoundError: No module named 'skbio.metadata._intersection'
```

The corresponding Makefile for that SHA is [here](https://github.com/biocore/scikit-bio/blob/cd7fedc6da5cea1b938e9a30b290a27f05a7b337/Makefile) and the Travis build is [here](https://travis-ci.org/biocore/scikit-bio/jobs/432673518).

Reply:
Looked into this with @ebolyen --- this is related to the `PYTHONPATH` jazz above (which, as you mentioned, is why we bother `cd`ing into `ci`). I was able to _kinda_ get this to work by hacking `sys.path`, but as @ebolyen says:

> the juice isn't worth the squeeze

🍋 🍋 🍋 

At the end of the day, `pytest-cov` is just wrapping `coverage.py`, so this sounds fine to "cut out the middle-person"